### ISSUE TITLE
Getting the share-screen prompt every view minutes in a wayland screen share session

### ISSUE THREAD

**Issue by amenk** [rustdesk/rustdesk#4276]
### Bug Description

I am using two screens

I am getting the prompt which screen to share every few minutes in a session

### How to Reproduce

Connect to my computer
wait

### Expected Behavior

Don't ask over and over again to share the screen

### Operating system(s) on local side and remote side

Ubuntu 22.04 -> Ubuntu 22.04

### RustDesk Version(s) on local side and remote side

fc4585b07e2108a01ed271a64241e051  rustdesk-1.2.0-x86_64.deb

### Screenshots

![image](https://user-images.githubusercontent.com/1087128/236407115-dd51ab4d-747d-4bd3-bc77-a85b1e8ab9ec.png)

**Comment by rustdesk** [comment]
@fufesou

**Comment by rustdesk** [comment]
@amenk can you try out obs, video meeting software? if they have this behavior.

**Comment by fufesou** [comment]
@amenk Hi, has your computer ever entered a lock screen state?
Screen sharing on Wayland does not support lock for now.

**Comment by amenk** [comment]
@rustdesk 
Microsoft Teams (teams-for-linux / in the browser) does not have the problem, OBS not yet tried. I don't have a TeamViewer license / and AnyDesk does not support Wayland at all (big plus for rustdesk) - would it help to try OBS as well?

@fufesou Yes, I am regularily locking my screen (but the prompt does not seem have a direct relation to that)

Any debug logs to check?

**Comment by rustdesk** [comment]
> locking my screen

Do you lock while you tested Teams?

**Comment by amenk** [comment]
@rustdesk yes I am locking my screen all the time when I leave the desk, not necessarily during a Teams session, but I also I did that before and never noticed this re-prompt.

Also I did not lock the screen in the current rustdesk session, and to get this right: it's not like I am locking my screen and then the re-prompt comes, the re-prompt just comes seemingly out of nowhere.

**Comment by rustdesk** [comment]
I do not think it is what we can control, that's also why anydesk etc does not support wayland.

**Comment by amenk** [comment]
Other than that, the wayland support works great :-)

**Comment by amenk** [comment]
@rustdesk what do you mean?

**Comment by rustdesk** [comment]
I will close this issue, because we can not find what we can do. Let's collect more feedback from the other users before dive into this.

**Comment by amenk** [comment]
I am also happy to help with debugging, are there debugging logs?

**Comment by rustdesk** [comment]
> I am also happy to help with debugging, are there debugging logs?

@fufesou

**Comment by amenk** [comment]
> Are you ok with current `circle` mouse cursor?

I just noticed that the remote side does not see any cursor .. but probably that is another issue :o)

**Comment by rustdesk** [comment]
> see any cursor .

Not implemented yet, because we have not found a good api to capture cursor image or name yet.

**Comment by fufesou** [comment]
> I am also happy to help with debugging, are there debugging logs?

@amenk Thanks.

The logs on the controlled side may be helpful.

Can you please package and send the log files to me (shuanglongchen@yeah.net) ?

The log file locations are:
```
~/.local/share/logs/RustDesk
/root/.local/share/logs/RustDesk
```

And you can also run `sudo journalctl -f`, to see if any logs are printed when the prompt occurs?
Send to me if you see any.

**Comment by amenk** [comment]
In the moment I only have the chance to try to reproduce with Android on the controlling side - so far it did not happen in this combination.
Will send logs if it happens again for example in the next session with a colleague.

I also have to say that I did not start the service at all in that past session with a colleague where it happend frequently (because I did not need any input control, only sharing). Maybe that might also be related.

I can trigger the re-prompt still with locking the screen, but this is fine.
What's also interesting: Whatever I choose there it has no effect. I have to displays (see also #4288) and initially shared the 2nd screen. Now when the re-prompt comes it does not make a difference if I chose 1st or 2nd. Still 2nd is shared.

I don't know how I even could switch to the 1st screen (probably I would need to delete the config). But that's maybe another issue :-)

But thanks again for this awesome project and the help so far.

**Comment by rustdesk** [comment]
Wayland is hard to maintain, as RustDesk developer, I failed to bring wayland up on Ubuntu 22.04 .1 (in VM) today, I remember it worked 2 months ago.

![image](https://user-images.githubusercontent.com/71636191/236616836-b36abfd1-46ef-4d2b-95dc-fd0ea36dc7d9.png)

Somehow xdg-desktop-portal has been removed, I install xdg-desktop-portal and xdg-desktop-portal-gnome again, but there is no screencast as above (run d-feet).

![image](https://user-images.githubusercontent.com/71636191/236617166-eb3c1127-e1e0-485e-b7d3-dc27b3c9fe95.png)

![image](https://user-images.githubusercontent.com/71636191/236616944-bd686954-20c2-4870-b4f4-c251529691dc.png)

https://github.com/flatpak/xdg-desktop-portal/issues/861

**Comment by rustdesk** [comment]
I have to close this issue because wayland screencast behavior is so unstable as stated above, it is totally out of our control.

**Comment by fufesou** [comment]
Logs from @amenk 

It seems that the capturer often fails to capture picture.
The capture and encoder are created in seconds.

> the reprompt happened today in a session between 17:33-19:00 quite ofen (approx all 5 minutes)
> My colleague is also on Ubuntu.

```
[2023-05-08 19:04:19.158033 +02:00] INFO [src/rendezvous_mediator.rs:179] request_pk received from rs-sg.rustdesk.com:21116
[2023-05-08 19:04:34.086344 +02:00] ERROR [src/server/service.rs:242] Error of video service: Wayland capturer none 100 times, try restart capture
[2023-05-08 19:04:34.146550 +02:00] DEBUG [src/server/service.rs:239] Enter video service inner loop
[2023-05-08 19:04:34.248396 +02:00] DEBUG [src/server/wayland.rs:143] #displays=1, current=0, origin: (1920, 0), width=3440, height=1440, cpus=8/8
[2023-05-08 19:04:34.252162 +02:00] INFO [src/server/wayland.rs:176] update mouse resolution: (1920, 5360), (0, 1440)
[2023-05-08 19:04:34.284413 +02:00] INFO [src/server/video_service.rs:457] init bitrate=4086, abr enabled:true
[2023-05-08 19:04:34.284465 +02:00] INFO [libs/scrap/src/common/codec.rs:99] new encoder:VPX(VpxEncoderConfig { width: 3440, height: 1440, timebase: [1, 1000], bitrate: 4086, codec: VP9, num_threads: 4 })
[2023-05-08 19:04:39.223719 +02:00] ERROR [src/server/service.rs:242] Error of video service: Wayland capturer none 100 times, try restart capture
[2023-05-08 19:04:39.283928 +02:00] DEBUG [src/server/service.rs:239] Enter video service inner loop
[2023-05-08 19:04:39.380816 +02:00] DEBUG [src/server/wayland.rs:143] #displays=1, current=0, origin: (1920, 0), width=3440, height=1440, cpus=8/8
[2023-05-08 19:04:39.383698 +02:00] INFO [src/server/wayland.rs:176] update mouse resolution: (1920, 5360), (0, 1440)
[2023-05-08 19:04:39.413298 +02:00] INFO [src/server/video_service.rs:457] init bitrate=4086, abr enabled:true
[2023-05-08 19:04:39.413350 +02:00] INFO [libs/scrap/src/common/codec.rs:99] new encoder:VPX(VpxEncoderConfig { width: 3440, height: 1440, timebase: [1, 1000], bitrate: 4086, codec: VP9, num_threads: 4 })
[2023-05-08 19:04:42.137110 +02:00] DEBUG [libs/hbb_common/src/udp.rs:35] Receive buf size of udp 0.0.0.0:0: Ok(212992)
[2023-05-08 19:04:56.144352 +02:00] DEBUG [src/rendezvous_mediator.rs:166] Latency of rs-sg.rustdesk.com:21116: 314.506ms
[2023-05-08 19:04:56.144446 +02:00] INFO [src/rendezvous_mediator.rs:179] request_pk received from rs-sg.rustdesk.com:21116
[2023-05-08 19:05:09.911044 +02:00] DEBUG [src/rendezvous_mediator.rs:166] Latency of rs-cn.rustdesk.com:21116: 320.385ms
[2023-05-08 19:05:12.658555 +02:00] ERROR [src/server/service.rs:242] Error of video service: Wayland capturer none 100 times, try restart capture
[2023-05-08 19:05:12.718760 +02:00] DEBUG [src/server/service.rs:239] Enter video service inner loop
[2023-05-08 19:05:12.793428 +02:00] DEBUG [src/server/wayland.rs:143] #displays=1, current=0, origin: (1920, 0), width=3440, height=1440, cpus=8/8
[2023-05-08 19:05:12.796937 +02:00] INFO [src/server/wayland.rs:176] update mouse resolution: (1920, 5360), (0, 1440)
[2023-05-08 19:05:12.827302 +02:00] INFO [src/server/video_service.rs:457] init bitrate=4086, abr enabled:true
[2023-05-08 19:05:12.827341 +02:00] INFO [libs/scrap/src/common/codec.rs:99] new encoder:VPX(VpxEncoderConfig { width: 3440, height: 1440, timebase: [1, 1000], bitrate: 4086, codec: VP9, num_threads: 4 })
[2023-05-08 19:05:22.371342 +02:00] ERROR [src/server/service.rs:242] Error of video service: Wayland capturer none 100 times, try restart capture
[2023-05-08 19:05:22.431561 +02:00] DEBUG [src/server/service.rs:239] Enter video service inner loop
[2023-05-08 19:05:22.526494 +02:00] DEBUG [src/server/wayland.rs:143] #displays=1, current=0, origin: (1920, 0), width=3440, height=1440, cpus=8/8
[2023-05-08 19:05:22.529570 +02:00] INFO [src/server/wayland.rs:176] update mouse resolution: (1920, 5360), (0, 1440)
[2023-05-08 19:05:22.559975 +02:00] INFO [src/server/video_service.rs:457] init bitrate=4086, abr enabled:true
[2023-05-08 19:05:22.560018 +02:00] INFO [libs/scrap/src/common/codec.rs:99] new encoder:VPX(VpxEncoderConfig { width: 3440, height: 1440, timebase: [1, 1000], bitrate: 4086, codec: VP9, num_threads: 4 })
```

**Comment by rustdesk** [comment]
reopen as @fufesou will follow up

**Comment by rustdesk** [comment]
@fufesou how about the status of this issue?

**Comment by fufesou** [comment]
I haven't delved into this issue recently, I'll take a look at it tomorrow.

**Comment by Aqua1ung** [comment]
> I haven't delved into this issue recently, I'll take a look at it tomorrow.

Please do! We're desperate!

**Comment by fufesou** [comment]
It may be my fault.
The most probable reason is there is no movement on the screen. There's no need to send new data.
I should not reset the capturer just in a few seconds.

https://github.com/rustdesk/rustdesk/blob/fb1207eee6970f8a58371b4b2b79757970683b9f/libs/scrap/src/wayland/pipewire.rs#L212

**Comment by rustdesk** [comment]
New build for x86-64 is ready.

**Comment by bayazidbh** [comment]
I know that this is merged, but portal seems to have merged remote-desktop-restore-token portal per late July, so it might be useful: flatpak/xdg-desktop-portal/pull/1004

**Comment by fufesou** [comment]
@bayazidbh Thanks. 

We've used this feature in our project. 😄

**Comment by TamaMcGlinn** [comment]
I just had this error with the latest release of RustDesk (1.2.3-1), on Ubuntu 22.04 LTS; is there any workaround?

**Comment by Mr-MooMoo** [comment]
Same error here   ; Trying to connect two  NixOS/Hyprland Hosts with  Client 1.2.4. ((Flutter))

**Comment by Ahrimdon** [comment]
> I just had this error with the latest release of RustDesk (1.2.3-1), on Ubuntu 22.04 LTS; is there any workaround?

Yes, the only work around I have found is by disabling Wayland.

In order to switch to X11:

1. Edit the gdm config: `sudo nano /etc/gdm3/custom.conf`

2. Uncomment `"WaylandEnable=false"` and `reboot`.

[Credit](https://askubuntu.com/a/1434620/1824372)
